tracer/src/Datadog.Trace/TracerManager.cs

```diff
@@ -216,6 +216,21 @@ private static async Task CleanUpOldTracerManager(TracerManager oldManager, Trac
         {
             try
             {
+                var telemetryReplaced = false;
+                if (oldManager.Telemetry != newManager.Telemetry && oldManager.Telemetry is not null)
+                {
+                    // This needs to be done as soon as possible (before awaiting anything) to avoid a race condition with TracerManager.Start
```

Comment:
Should this be done in `ReplaceGlobalManager` then? as `CleanUpOldTracerManager` is fire and forget, isn't there a tiny tiny possibility we start the new manager before we set the flag?

Reply:
Well that's why I'm doing it in the synchronous part of the method. Before the first await the execution of `ReplaceGlobalManager` won't resume.
I could do it in `ReplaceGlobalManager` but it feels eve more dirty to me 🤔

Reply:
Oh never mind, I can do it in `CreateTracerManager`, that's much better